src/compression/lz4.rs

```diff
@@ -0,0 +1,502 @@
+use std::mem;
```

Comment:
Sure, I write it.

In fact, I have extracted it to an isolated crate [lz4frame](https://github.com/flier/lz4frame). I'm not sure you like to depend on external crate or embedded those code.

[lz4-rs](https://github.com/bozaro/lz4-rs) have similar LZ4 frame supports from `liblz4`, but it doesn't support the wrong frame checksum used in Kafka 0.8.